hostfactory/host_provider/src/cyclecloud_provider.py

```diff
@@ -1232,34 +930,47 @@ def debug_completed_nodes(self):
                 internal_completed_nodes.update(set(request.get("completedNodes")))
         incomplete_nodes = actual_completed_nodes - internal_completed_nodes
         print(incomplete_nodes)
-
-def bucket_priority(buckets, bucket_nodearray):
-    nodearrays = list(dict.fromkeys(bucket.nodearray for bucket in buckets))
-    filter_bucket_nodearray = [bucket for bucket in buckets if bucket.nodearray == bucket_nodearray.nodearray]
-    b_index = next((i for i, bucket in enumerate(filter_bucket_nodearray) if bucket.bucket_id == bucket_nodearray.bucket_id), None)
-    reveresed_nodearrays = nodearrays[::-1]
-    return _bucket_priority(reveresed_nodearrays, bucket_nodearray, b_index)
-
-def _bucket_priority(nodearrays, bucket_nodearray, b_index):
-    prio = bucket_nodearray.priority
-    if isinstance(prio, str):
-        try:
-            prio = int(float(prio))
-        except Exception:
-            prio = None
-    if isinstance(prio, float):
-        prio = int(prio)
-    if isinstance(prio, int):
-        if prio < 0:
-            prio = None
-    if prio is not None and not isinstance(prio, int):
-        prio = None    
-    if prio is None:
-        prio = (nodearrays.index(bucket_nodearray.nodearray) + 1) * 10
-    if prio > 0:
-        return prio * 1000 - b_index
-    assert prio == 0, f'Unexpected prio {prio} - should ALWAYS be >= 0.'
-    return prio   
+    
+    def validate_template(self):
+        cluster_status = self.cluster.status()
+        nodearrays = cluster_status["nodearrays"]
+        pro_conf_dir=os.getenv('PRO_CONF_DIR', os.getcwd())
+        conf_path = os.path.join(pro_conf_dir, "conf", "azureccprov_templates.json")
+        with open(conf_path, 'r') as json_file:
+            templates_json = json.load(json_file)
+        if "templates" not in templates_json:
+            print("List templates not present in azureccprov_templates.json")
+            return False
+        templates_json = templates_json["templates"]
+        if len(templates_json) == 0:
+            print("Length of list templates is 0")
+            return False
+        for template in templates_json:
+           for nodearray_root in nodearrays:
+               template_name_found = False
```

Comment:
I think this should be initialized outside of the for loop of nodearrays? I am getting a "this variable may not have been initialized" warning on line 967

Reply:
yes will initialize outside loop true